core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -74,11 +93,20 @@ class MapUnit {
         private const val ANCIENT_RUIN_MAP_REVEAL_CHANCE = 0.8f
     }
 
+    // GDX serialization needs a no-arg constructor to be defined, if any constructor is.
+    constructor() {}
+
+    constructor(baseUnit_ : BaseUnit, civInfo_: CivilizationInfo) {
```

Comment:
In what situation is this actually necessary? It seems that you've changed a lot of code to fit this patters, but it's unclear why the pattern is needed at all. In any case this change (and all related changes) should be part of a separate PR.

Reply:
This is kind-of a holdover from an early change I tried to remove `name` altogether, before I came to understand that it needed to exist and be publically settable in order for deserialization to work. 

I still have some concern about the possibility of  `baseUnit` and `name`  or `civInfo` and `owner` getting out of sync, but you're right that's an issue for another PR.